modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -205,12 +207,8 @@ func (t TLSConfig) MakeTLSClientConfig() (*tls.Config, error) {
 	// trusted root CAs
 	if len(t.RootCAPool) > 0 {
 		rootPool := x509.NewCertPool()
-		for _, encodedCACert := range t.RootCAPool {
-			caCert, err := decodeBase64DERCert(encodedCACert)
-			if err != nil {
-				return nil, fmt.Errorf("parsing CA certificate: %v", err)
-			}
-			rootPool.AddCert(caCert)
+		for _, caCert := range t.RootCAPool {
+			rootPool.AppendCertsFromPEM([]byte(caCert))
```

Comment:
This is comparable to the approach over here: 
https://github.com/caddyserver/caddy/blob/5ab17a3a371c1f84ef5a649d9310971eb7959271/modules/caddytls/acmemanager.go#L130-L142

We could adapt `t.RootCAPool` to also just be the file paths as well, either way this change would be incompatible with any existing v2 configs that were putting the pem block directly in the config. Still I'd argue this is the right thing to do while v2 is still in beta.

Reply:
Assuming the functionality of putting a pem block directly in the config is there because it was required by someone, how can the server maintain backwards compatibility? I hate adding flags/switches but it seems your needs are a different requirement or option someone may want to leverage?

Reply:
Given the inconsistency with the acmemanager it self, it seems worthwhile to validate the specific requirement to load the pem block directly. Given that v2 is still in beta, I'd push to consider that this _is the time_ for backwards incompatible changes. Loading the pem block directly seems a bit usual to begin with IMO, the only context I've seen that before is when building keys directly into binaries to make it less obvious that it's a cert, which doesn't apply here.

That said, if we wanted to support both, I'd suggest we add an addition field here for files, and leave the pem block logic as it is

https://github.com/caddyserver/caddy/blob/v2/modules/caddyhttp/reverseproxy/httptransport.go#L176


Then we can follow the same approach shown in `caddy/modules/caddytls/acmemanager.go`, but again given v2's status as a beta, this is the best time to standardize the approach.

Reply:
I get this is in beta but I don't want to use that as a blank check to break others.

Right now Matt is the only developer on the project and he has some high priority items he needs to get done by mid January. If you can make a non-breaking change that is reasonable, it will be easier to make a decision to merge.

This being open source, you could fork the project for now, make your change and then early next year we can attempt to bring those changes back in.